pybamm/models/full_battery_models/lithium_ion/tanks_in_series.py

```diff
@@ -0,0 +1,289 @@
+        c_12 = (
+            eps_p**param.p.b_e / param.p.L * c_e_p
+            + eps_sep**param.s.b_e / param.s.L * c_e_s
+        ) / (eps_p**param.p.b_e / param.p.L + eps_sep**param.s.b_e / param.s.L)
+        # 23: boundary between separator and negative electrode
+        c_23 = (
+            eps_n**param.n.b_e / param.n.L * c_e_n
+            + eps_sep**param.s.b_e / param.s.L * c_e_s
+        ) / (eps_n**param.n.b_e / param.n.L + eps_sep**param.s.b_e / param.s.L)
+
+        # Grouped thickness/porosity variables
+        # positive electrode and separator
+        leps12 = param.p.L / eps_p**param.p.b_e + param.s.L / eps_sep**param.s.b_e
+        # negative electrode and separator
+        leps23 = param.n.L / eps_n**param.n.b_e + param.s.L / eps_sep**param.s.b_e
+
+        ######################
+        # Electrolyte concentration
+        ######################
+
+        self.rhs[c_e_n] = (
+            -2 * param.D_e(c_23, T) * (c_e_n - c_e_s) / leps23 / (eps_n * param.n.L)
+            - (1 - param.t_plus(c_e_n, T)) * iapp / param.F / eps_n / param.n.L
+        )
+        self.rhs[c_e_s] = (
+            -2 * param.D_e(c_12, T) * (c_e_s - c_e_p) / leps12
+            + 2 * param.D_e(c_23, T) * (c_e_n - c_e_s) / leps23
+        ) / (eps_sep * param.s.L)
+        self.rhs[c_e_p] = (
+            -2 * param.D_e(c_12, T) * (c_e_s - c_e_p) / leps12 / (eps_p * param.p.L)
+            - (1 - param.t_plus(c_e_p, T)) * iapp / param.F / eps_p / param.p.L
+        )
+
+        self.initial_conditions[c_e_n] = param.c_e_init
+        self.initial_conditions[c_e_s] = param.c_e_init
+        self.initial_conditions[c_e_p] = param.c_e_init
+
+        ######################
+        # Algebraic Variables
+        ######################
+
+        ######################
+        # Average Surface Concentration
+        ######################
+
+        self.algebraic[c_surf_ave_p] = (
+            j_p
+            + 35
+            * param.p.prim.D(c_s_p, T)
+            / param.p.prim.R_typ
+            * (c_surf_ave_p - c_s_p)
+            - 8 * param.p.prim.D(c_s_p, T) * q_ave_p
+        )
+        self.algebraic[c_surf_ave_n] = (
+            j_n
+            + 35
+            * param.n.prim.D(c_s_n, T)
+            / param.n.prim.R_typ
+            * (c_surf_ave_n - c_s_n)
+            - 8 * param.n.prim.D(c_s_n, T) * q_ave_n
+        )
+
+        self.initial_conditions[c_surf_ave_p] = param.p.prim.c_init
+        self.initial_conditions[c_surf_ave_n] = param.n.prim.c_init
+
+        ######################
+        # Current in the electrolyte
+        ######################
+
+        self.algebraic[phi_e_p] = (
+            -iapp
+            - 2 * param.kappa_e(c_12, T) * (phi_e_s - phi_e_p) / leps12
+            + 4
+            * param.R
+            * T
+            * param.F
+            * param.chi(c_12, T)
+            / 2
+            * param.kappa_e(c_12, T)
+            / c_12
+            * (c_e_s - c_e_p)
+            / leps12
```

Comment:
grouping things in brackets, or precomputing some things (e.g. `RT_F = param.R * T / param.F`) will make this easier to read